app/api/models/genome.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from typing import Optional
+
+from pydantic import BaseModel, Field, validator, AliasChoices, AliasPath
+
+from core.logging import InterceptHandler
+from core.config import ENA_ASSEMBLY_URL
+
+logging.getLogger().handlers = [InterceptHandler()]
+
+
+class Type(BaseModel):
+    kind: str
+    value: str
+
+
+class AssemblyInGenome(BaseModel):
+    accession_id: str = Field(alias='accession')
+    name: str = Field(alias='name')
+    url: str = Field(alias='accession', default=None)
+
+    @validator("url", always=True)
+    def generate_url(cls, value):
+        if value and "GCA" in value:
+            return ENA_ASSEMBLY_URL + value
+        return None
+
+
+class AssemblyProvider(BaseModel):
+    name: str
+    url: str
+
+
+class AnnotationProvider(BaseModel):
+    name: str
+    url: str
+
+
+class GenomeDetails(BaseModel):
+    genome_id: str = Field(alias='genomeUuid')
+    genome_tag: Optional[str] = Field(
+        alias=AliasChoices(AliasPath('assembly', 'urlName'), AliasPath('assembly', 'tolId')), default=None)
+    taxonomy_id: str = Field(alias=AliasPath('organism', 'taxonomyId'))
+    species_taxonomy_id: str = Field(alias=AliasPath('organism', 'speciesTaxonomyId'))
```

Comment:
Probably for @azangru  Why is species_taxonomy_id string here and same is int in popular_species endpoint ? for backend do we want consistency ?

Reply:
> Probably for @azangru Why is species_taxonomy_id string here and same is int in popular_species endpoint ? for backend do we want consistency ?

You might remember that I actually asked BE this question, wondering whether you want to be consistent across your apis and endpoints 🙂 For FE, it does not matter one way or the other. It probably makes more sense to have those ids consistently as strings, because you don't do arithmetic operations with them; and because the fact that they consist only of digits is purely accidental; but FE can deal with numbers and strings equally easily. We turn them into strings when we render them on the page or include them in url parameters anyway.

Reply:
If we not it is always going to be int then this should be int. It will also ensure integrity and on less step for validation.  I will leave it to @Mehrnaz-Charkhchi

Reply:
I remember the discussion that @azangru is referring to. We went with string since we were getting all of these values as string from EBI search. If we want to change this we have to update both repos specs and do this conversion in in search hub repo.

Reply:
Will have to make changes to popular species endpoint to make it str